process/coordinator/process.go

```diff
@@ -1673,28 +1676,19 @@ func (tc *transactionCoordinator) getMaxAccumulatedAndDeveloperFees(
 	maxAccumulatedFeesFromMiniBlock := big.NewInt(0)
 	maxDeveloperFeesFromMiniBlock := big.NewInt(0)
 
-	miniBlockHash, err := core.CalculateHash(tc.marshalizer, tc.hasher, miniBlock)
+	pi, err := tc.getIndexesOfLastTxProcessed(miniBlock, processedMiniBlocks, miniBlockHeaderHandler)
 	if err != nil {
 		return big.NewInt(0), big.NewInt(0), err
 	}
 
-	indexOfLastTxProcessedByItself := int32(-1)
-	if processedMiniBlocks != nil {
-		processedMiniBlockInfo, _ := processedMiniBlocks.GetProcessedMiniBlockInfo(miniBlockHash)
-		indexOfLastTxProcessedByItself = processedMiniBlockInfo.IndexOfLastTxProcessed
+	indexOfFirstTxToBeProcessed := pi.indexOfLastTxProcessedByItself + 1
+	err = process.CheckIfIndexesAreOutOfBound(indexOfFirstTxToBeProcessed, pi.indexOfLastTxProcessedByProposer, miniBlock)
+	if err != nil {
+		return big.NewInt(0), big.NewInt(0), err
```

Comment:
we return 2 non nil values and a non nil error. Is this ok?

Reply:
Actually this is the pattern used in case of an error also below in L1695 in the old code. But I checked in the caller side, called only from one place, and these values are not used further in case of an error. The caller method returns only the error and these values are dropped in that point. So, I changed them to nil, nil.
```
		maxAccumulatedFeesFromMiniBlock, maxDeveloperFeesFromMiniBlock, err := 
                        tc.getMaxAccumulatedAndDeveloperFees(
			header.GetMiniBlockHeaderHandlers()[index],
			miniBlock,
			mapMiniBlockTypeAllTxs[miniBlock.Type],
			processedMiniBlocks,
		)
		if err != nil {
			return err
		}

```